### PR TITLE
Sanitize the appId passed to `findAppInDirectories`

### PR DESCRIPTION
**Pull request by LukasReschke**
Would have prevented quite some security bugs in the past. Nice hardening for now.

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
:+1:

File: lib/private/app.php [hunk 1]
```diff
@@ -474,9 +474,13 @@ public static function getInstallPath() {
 	 * search for an app in all app-directories
 	 *
 	 * @param string $appId
-	 * @return mixed (bool|string)
+	 * @return false|string
```
Comment (rullzer) [inline lib/private/app.php]: False is not a type...

Comment (LukasReschke) [inline lib/private/app.php]: It's perfectly legit. http://www.phpdoc.org/docs/latest/guides/types.html

Comment (LukasReschke) [inline lib/private/app.php]: > a special keyword specific to the PHPDoc Standard, such as false or mixed.

**Comment by rullzer** [comment]
Besides my nit picking... which you can ignore :P

:+1:

**Pull request merged**